pkg/controller/machineset_test.go

```diff
@@ -871,7 +871,180 @@ var _ = Describe("machineset", func() {
+							Phase: MachineRunning,
+						},
+					},
+				}
+
+				objects := []runtime.Object{}
+				objects = append(objects, testMachineSet, testActiveMachine1, testActiveMachine3)
+				c, trackers := createController(stop, testNamespace, objects, nil, nil)
+				defer trackers.Stop()
+				waitForCacheSync(stop, c)
+
+				machines, _ := c.controlMachineClient.Machines(testNamespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(len(machines.Items)).To(Equal(int(testMachineSet.Spec.Replicas) - 1))
+
+				activeMachines := []*machinev1.Machine{testActiveMachine1, testActiveMachine3}
+				Err := c.manageReplicas(context.TODO(), activeMachines, testMachineSet)
+				waitForCacheSync(stop, c)
+				machines, _ = c.controlMachineClient.Machines(testNamespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(len(machines.Items)).To(Equal(int(testMachineSet.Spec.Replicas)))
+				Expect(Err).Should(BeNil())
+			})
+
+			// Testcase: ActiveMachines + MachinesWithUpdateSuccessfulLabel = DesiredMachines
+			// It should not return error.
+			It("should not create or delete machines and should not return error", func() {
+				stop := make(chan struct{})
+				defer close(stop)
+
+				testActiveMachine3 = &machinev1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "machine-3",
+						Namespace: testNamespace,
+						UID:       "12345610",
+						Labels: map[string]string{
+							"test-label":                             "test-label",
+							"node.machine.sapcloud.io/update-result": "successful",
+						},
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Machine",
+						APIVersion: "machine.sapcloud.io/v1alpha1",
+					},
+					Status: machinev1.MachineStatus{
+						CurrentStatus: machinev1.CurrentStatus{
+							Phase: MachineRunning,
+						},
+					},
+				}
+
+				objects := []runtime.Object{}
+				objects = append(objects, testMachineSet, testActiveMachine1, testActiveMachine2, testActiveMachine3)
+				c, trackers := createController(stop, testNamespace, objects, nil, nil)
+				defer trackers.Stop()
+				waitForCacheSync(stop, c)
+
+				machines, _ := c.controlMachineClient.Machines(testNamespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(len(machines.Items)).To(Equal(int(testMachineSet.Spec.Replicas)))
+
+				activeMachines := []*machinev1.Machine{testActiveMachine1, testActiveMachine2, testActiveMachine3}
+				Err := c.manageReplicas(context.TODO(), activeMachines, testMachineSet)
+				waitForCacheSync(stop, c)
+				machines, _ = c.controlMachineClient.Machines(testNamespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(len(machines.Items)).To(Equal(int(testMachineSet.Spec.Replicas)))
+				Expect(Err).Should(BeNil())
+			})
+
+			// TestCase: ActiveMachines + MachinesWithUpdateSuccessfulLabel > DesiredMachines
+			// Testcase: It should not return error and delete extra running machine.
+			It("should not create or delete machines and should not return error", func() {
```

Comment:
Please update the subject node description in accordance with the expectation and remove the above comment line
```suggestion
			It("should not return error and delete extra running machine", func() {
```